Whitney

Whits26 opened the pull request:
Implemented sign up working
Models created

Whits26 committed:
Admin Model

Whits26 committed:
Inbox Model

Whits26 committed:
Post Model

Whits26 committed:
Topic Model Update

Whits26 committed:
PostTag Model

Whits26 committed:
Model

Whits26 committed:
Inbox Model

Whits26 committed:
Post Model

Whits26 committed:
PostTag Model

Whits26 committed:
Reply model

Whits26 committed:
Tag Model

Whits26 committed:
Topic Model

Whits26 committed:
Manage Update of working Admin Sign Up

Whits26 committed:
User controller update sign up working

Snickdx commented on manage.py:
```diff
@@ -65,5 +68,25 @@ def createAdmin(fname="robAdmin", lname="Smith", email="robadmin@mail.com", pass
+    p1 = input('Enter ADMIN password :')
+    print (p1)
+
+    fname1 = User(first_name=n1)
```
lines 83-84 creates 4 user objects and does nothing with it?

Whits26 replied:
I have fixed this, hope this is correct.

    User (first_name = n1)
    User (last_name = n2)
    User (email = e1)
    User (password = p1)

Will make changes to GitHub once I have finished the other corrections

Snickdx commented on App/controllers/user.py:
```diff
@@ -1,8 +1,11 @@
-from flask import render_template
+from flask import redirect, render_template, request, url_for, jsonify
```
controllers should never redirect, render_template, use request, URL or jsonify as these are operations that should remain on the view

Whits26 replied:
I think I have fixed this now, I have tested it and it works

CONTROLLER
# REGISTER A NORMAL USER
def register(fname,lname,email,password):
    publicUser = User(first_name=fname, last_name=lname, email=email, password=password)
    publicUser.set_password(password)
    db.session.add(publicUser)
    db.session.commit()

VIEW
# REGISTRATION NORMAL USER
@user_views.route('/register', methods=['GET','POST'])
def get_user_reg_page():
    if request.method == 'POST':
        try:
            fname = request.form['first_name']
            lname = request.form['last_name']
            email = request.form['email']
            password = request.form['password']
            RegisterUser = register(fname,lname,email,password)
        except IntegrityError:
            db.session.rollback()
            return jsonify('Something went wrong. User NOT Registered')
        return jsonify('User Registered')
        
    return render_template('register.html')

Snickdx reviewed (changes requested):
User controller has view level dependencies; request, jsonify, redirect and render_template, these should never be used in a controller

Whits26 committed:
Manage.py Update Admin Sign up

Whits26 committed:
User Updated Controller Sign Up

Whits26 committed:
User Updated View Sign Up

Snickdx committed:
updated manage.py

createAdmin was already defined, renamed the to makeAdmin, models should not be referenced directly in commands, use a controller instead, use more meaningful variable names, commands return values are not used

Snickdx reviewed (approved):
reviewed
